src/Email/Email.php

```diff
@@ -123,8 +65,6 @@ public function __construct(array $props = [], bool $debug = false)
 
 	/**
 	 * Returns the email attachments
-	 *
-	 * @return array
 	 */
 	public function attachments(): array
```

Comment:
All of these getter methods except `isSent()` can currently also return `null` because the props are all nullable.

Maybe we clean the prop types a bit by updating the setters to replace `null` with an empty array in the `setAttachments()`, `setBcc()`, `setCc()`, `setTo()` and `setTransport()` methods. The getters of the remaining props need to be updated to return nullable values.

Reply:
Can they? `setAttachments` already ensure that an empty array is set if `null` is passed. Calling `resolveEmail()` in the setters also does that in the setters for bcc, cc, too. `::transport()` returns `::defaultTransport()` when it's null.

Type hinting at the top needs to include null as otherwise PHP throws an error, which I think is due to our `Properties` trait magic - not that the properties could actually be null once the constructor and magic has finished.

Reply:
You are right. Then I think we should actually `@var` comments to the props again to mark the actual possible types after the constructor has run. Otherwise we will be confused down the line.

Reply:
Not sure if this would be less confusing (once you forget which one is the right one). I think the actual path forward is getting rid of the Properties trait.

Reply:
Reconsidered, `@var` comments might be good. Adding them.